src/swap/SwapScreenV2.tsx

```diff
@@ -0,0 +1,1149 @@
+        ? { networkId: selectedToken.networkId, field: Field.TO }
+        : null
+
+    setSwitchedToNetworkId(allowCrossChainSwaps ? null : newSwitchedToNetwork)
+
+    trackConfirmToken({
+      field: Field.TO,
+      selectedToken,
+      newFromToken: newSwitchedToNetwork ? undefined : fromToken,
+      newToToken: selectedToken,
+      newSwitchedToNetworkId: newSwitchedToNetwork?.networkId ?? null,
+      tokenPositionInList,
+    })
+
+    if (selectedToken?.tokenId !== toToken?.tokenId) {
+      setStartedSwapId(undefined)
+    }
+
+    if (newSwitchedToNetwork) {
+      // reset the fromToken if the user is switching networks
+      setFromToken(undefined)
+      clearQuote()
+    }
+
+    /**
+     * Use requestAnimationFrame so that the bottom sheet and keyboard dismiss
+     * animation can be synchronised and starts after the state changes above.
+     * Without this, the keyboard animation lags behind the state updates while
+     * the bottom sheet does not.
+     */
+    requestAnimationFrame(() => tokenBottomSheetToRef.current?.close())
+  }
+
+  function handleSelectAmountPercentage(percentage: number) {
+    handleSelectPercentageAmount(percentage)
+    setSelectedPercentage(percentage)
+
+    if (!fromToken) {
+      // Should never happen
+      return
+    }
+    AppAnalytics.track(SwapEvents.swap_screen_percentage_selected, {
+      tokenSymbol: fromToken.symbol,
+      tokenId: fromToken.tokenId,
+      tokenNetworkId: fromToken.networkId,
+      percentage,
+    })
+  }
+
+  function onPressLearnMore() {
+    AppAnalytics.track(SwapEvents.swap_learn_more)
+    navigate(Screens.WebViewScreen, { uri: links.swapLearnMore })
+  }
+
+  function onPressLearnMoreFees() {
+    AppAnalytics.track(SwapEvents.swap_gas_fees_learn_more)
+    navigate(Screens.WebViewScreen, { uri: links.transactionFeesLearnMore })
+  }
+
+  return (
+    <SafeAreaView style={styles.safeAreaContainer} testID="SwapScreen">
+      <CustomHeader
+        style={{ paddingHorizontal: variables.contentPadding }}
+        left={<BackButton />}
+        title={t('swapScreen.title')}
+      />
+      <ScrollView
+        contentContainerStyle={styles.contentContainer}
+        keyboardShouldPersistTaps="handled"
+      >
+        <View style={{ display: 'flex', justifyContent: 'space-between', flexGrow: 1 }}>
+          <View style={{ flexShrink: 0 }}>
+            <View style={[styles.warningsContainer, { gap: 4 }]}>
+              <TokenEnterAmount
+                autoFocus
+                token={fromToken}
+                inputValue={amountFrom}
+                inputRef={inputFromRef}
+                tokenAmount={processedAmountsFrom.token.displayAmount}
+                localAmount={processedAmountsFrom.local.displayAmount}
+                onInputChange={handleAmountInputChange}
+                amountType={amountType}
+                toggleAmountType={handleToggleAmountType}
+                onOpenTokenPicker={onOpenTokenPickerFrom}
```

Comment:
supernit: shall we prefix this handler with `handle-` instead of `on-` naming for uniformity?

Reply:
@bakoushin I think it makes sense to keep props of components with `on` to follow the regular naming convention of event handlers for components (e.g. `onClick`/`onBlur` etc). 

Naming function handlers with `handle` totally makes sense to me though. Will change that!

Reply:
@bakoushin I think it makes sense to keep props of components with `on` to follow the regular naming convention of event handlers for components (e.g. `onClick`/`onBlur` etc). 

Naming function handlers with `handle` totally makes sense to me though. Will change that!